collector/client.go

```diff
@@ -38,55 +28,36 @@ type SquidClient interface {
 	GetInfos() (types.Counters, error)
 }
 
-const (
-	requestProtocol = "GET cache_object://localhost/%s HTTP/1.0"
-)
-
-func buildBasicAuthString(login string, password string) string {
-	if len(login) == 0 {
-		return ""
-	} else {
-		return base64.StdEncoding.EncodeToString([]byte(login + ":" + password))
-	}
-}
-
-type CacheObjectRequest struct {
-	Hostname string
-	Port     int
-	Login    string
-	Password string
-	Headers  []string
-}
-
 /*NewCacheObjectClient initializes a new cache client */
-func NewCacheObjectClient(cor *CacheObjectRequest) *CacheObjectClient {
+func NewCacheObjectClient(hostname string, port int, username, password string, headers []string) *CacheObjectClient {
```

Comment:
Is there any reason switching from the request object to multiple arguments? 

For the context: I did this change a while ago to avoid changing the function signature when introducing new arguments (or removing one).

Reply:
Due to the age of this PR, I may have forgotten the exact reasons, but explicitly listing the params in the function signature makes the code easier to read and is more in line with traditional Go programming style. I think it's unlikely that this function signature will change often (if ever), since the current set of params pretty much covers all requirements. Also, despite `NewCacheObjectClient` being a publicly exported function, are there actually any downstream projects using this code as a library? AFAICT, it is only an exported function since it invoked by the `main` package of this exporter.

Reply:
I'm not aware if anyone uses this method in other projects, but I think I changed this function in the past a few times, so it's not very unlikely that it's going to change. Almost any new configuration may introduce a new parameter. I'll be happy to keep it this way, but I think we should changing it back if we ever need to change this signature again.